Geospatial/Travel Times/travel times functions.R

```diff
@@ -0,0 +1,74 @@
+# get_travel_areas - Function to obtain the area that is travellable from location (longitude, latitude) within
+#                    a specified travel_time using a particular travel_method.
+# - longitude     - vector of longitudes of locations being travelled from
+# - latitude      - vector of latitudes of locations being travelled from
+# - travel_time   - Travel time of interest in minutes. Recommend that this be no less than 10
+# - travel_method - Method of travel. One of "car", "bike", or "foot"
+# - within_region - Optional region that the travel area should remain within. object of class sf
+
+get_travel_areas <- function(longitude, latitude, travel_time, travel_method = c("car", "bike", "foot"), within_region = NULL) {
+  travel_method <- rlang::arg_match(travel_method)
+
+  # Get travel areas
+  iso <- purrr::map2(
+    longitude, latitude, # Iterate over each point (slow)
+    \(x, y) get_iso(x, y, travel_time, travel_method)
+  )
+
+  # remove empty areas - generally happens when travel time is set < 10
+  iso <- iso[purrr::map_int(iso, nrow) != 0]
+
+  # Combine into a single area to make plotting easier
+  # suppress warnings related to assumptions being made about spatial attributes
+  # there is one wanring for each area merged which isn't relevant as all areas are generated using the same crs
+  iso <- suppressWarnings(purrr::reduce(iso, \(x, y) st_union(st_make_valid(x), y)))
+
+  # Remove odd small regions
+  # [[1]] subsetting needed as the geometry column is a list with one entry because of st_union()
```

Comment:
I looked at `unlist()` and it either flattens too much so that the actual data structure within it is flattened to a vector or doesn't flatten enough, so still has a list element. I could use `pluck(iso$geometry, 1)` but that's not all that different to just the indexing that I've used. `list_flatten()` doesn't work because although it's a list it still inherits spatial properties so it throws an error. If you've any other suggestions I'm happy to hear them - I haven't used `purrr` a huge amount especially the more niche functions.

Reply:
It needs to be within the if statement otherwise it will lose the geography. Most recent commit has an option using `unlist()` with `recursive = FALSE` in two separate places to keep things in the format they need to be to get them back in to something spatial.  
```suggestion
  if (inherits(iso$geometry, "sfc_MULTIPOLYGON")) {
    iso_geometry <- unlist(iso$geometry, recursive = FALSE)
    small_area <- purrr::map_lgl(unlist(iso_geometry, recursive = FALSE), \(x) nrow(x) <= 5)
    st_geometry(iso) <- st_sfc(st_multipolygon(iso_geometry[!small_area]), crs = 4326) # Add the polygons together to make mapping simpler
  }
```